packages/next/client/head-manager.ts

```diff
@@ -58,25 +79,32 @@ function updateElements(
       if (title !== document.title) document.title = title
       return
     }
+    // handle react components as a function, functional component, or a class component
+    if (typeof tag.type === 'function') {
+      reactComponentToDOM(tag).forEach((t) => newTags.add(t))
+    } else {
+      newTags.add(reactElementToDOM(tag))
+    }
```

Comment:
Aren't we going to go through `newTags` for each component we update? That doesn't seem right. We probably want `newTags` to be local to this loop, right?

As an aside, I'd prefer if `reactElementToDOM` could just handle the component case, so we can remove the branch here and just deal with an array of elements.